DOC-5330 - HCD migration toolkit page (#207)
datastax/migration-docs: modules/ROOT/pages/hcd-migration-paths.adoc

Whenever possible, {company} strongly recommends using the {product} ({product-short}) tools to orchestrate ongoing read/write traffic when you migrate to {hcd-short}.

[#zdm-to-hcd]
== Zero-downtime migrations to {hcd-short}

The {product} ({product-short}) tools allow you to standup your new {hcd-short} clusters independently of your existing {dse-short} clusters.
Then, {product-proxy} orchestrates live traffic and synchronizes ongoing writes while you migrate data to your new clusters using any {product-short}-compatible data migration and validation tool.
Finally, you can use {product-proxy} to simulate the live workload on your new clusters before permanently switching your traffic over.

{product-proxy} and {product-automation} provide the safest upgrade approach with blue-green deployment capabilities that eliminate time pressure and ensure optimal availability and operational safety.
You can rollback up to the last stage of the migration if necessary.

By orchestrating independent clusters with the {product-short} tools, you can specify your ideal {hcd-short} configuration settings that you otherwise wouldn't be able to change during an in-place cluster upgrade.

> Incompatibilities in cluster configuration don't disrupt the migration because your existing cluster stays active and unchanged while you setup the new cluster and migrate your data.

Reviewer: ```suggestion
Incompatibilities in cluster configuration don't disrupt the migration because your existing cluster remains active and unchanged while you set up the new cluster and migrate your data.
```

=== Data validation with {product-short}

The {product-short} tools don't migrate your data.
During the {product-short} process, you use a xref:ROOT:migrate-and-validate-data.adoc[data migration tool] to rewrite the data from your existing cluster to your new cluster.

{company} recommends that you do the following:

* Choose a data migration tool that also includes strong validation capabilities, such as [{cass-migrator} ({cass-migrator-short})].
* Be aware of incompatible data types that can fail to migrate from your old cluster.

> For example, {hcd-short} 1.2 doesn't support tuples in {dse-short} versions 6.8.4 and earlier.

Reviewer: ```suggestion
For example, {hcd-short} 1.2.3 doesn't support tuples in {dse-short} versions 6.8.4 and earlier.
```

Reviewer: Versioning got complicated with internal releases and the IBM preview GA. GA for "1.2" will be 1.2.3.

> Data validation tools can catch these inconsistencies as missing or mismatched data, but you will still need to have a plan to resolve the inconsistencies.

Reviewer: ```suggestion
Data validation tools can identify these inconsistencies as missing or mismatched data, but you will still need a plan to resolve them.
```

> For example, you might need to modify your applications to use a different data type, or you might need to perform additional post-migration writes to populate lost data.

Reviewer: ```suggestion
For example, you might need to modify your applications to use a different data type or perform additional post-migration writes to populate lost data.
```

It is crucial that you fully validate and test your new cluster before switching your traffic over to it.

{product-proxy} is ideal for supporting this transition because it allows both clusters to remain in place until you are completely certain you are ready to switch to the new cluster.
Additionally, your old cluster remains untouched and available for rollback or reversion if necessary.

=== Get started with {product-short} and {hcd-short}

For information about clusters that are eligible for {product} to {hcd-short}, see xref:ROOT:zdm-proxy-migration-paths.adoc[].

To begin your {product} to {hcd-short}, go to xref:ROOT:introduction.adoc[].

> You must setup your {hcd-short} clusters before you can enable the {product-proxy}.

Reviewer: ```suggestion
You must set up your {hcd-short} clusters before you can enable the {product-proxy}.
```
